packages/babel-types/test/validators.js

```diff
@@ -32,6 +32,52 @@ suite("validators", function() {
     });
   });
 
+  suite("isCompatTag", function() {
+    it("should handle lowercase tag names", function() {
+      assert(t.react.isCompatTag("div"));
+      assert(t.react.isCompatTag("a")); // one letter
+      assert(t.react.isCompatTag("h3")); // letters and numbers
+    });
+
+    it("should handle custom element tag names", function() {
+      assert(t.react.isCompatTag("plastic-button")); // ascii letters
+      assert(t.react.isCompatTag("math-α")); // non-latin chars
+      assert(t.react.isCompatTag("img-viewer2")); // numbers
+      assert(t.react.isCompatTag("emotion-😍")); // emoji
```

Comment:
Not sure if anything besides `/^[a-z][a-z-]*$/` is supposed to work, but I don't see why ban it unless the JSX spec forbids it.

Reply:
Checked the spec; the spec actually says nothing about compat tags at all; just that the identifier (`JSXIdentifier`) is basically the same thing as [`IdentifierName`](https://tc39.github.io/ecma262/#prod-IdentifierName) but it also allows `-`.

I assume that that part is enforced by babylon instead of `isCompatTag` so this seems sufficient.

Reply:
Emojis in source code are cool :laughing: